lighthouse-core/gather/gatherers/viewport-dimensions.js

```diff
@@ -21,38 +21,44 @@ const Gatherer = require('./gatherer');
 /* global window */
 
 /* istanbul ignore next */
-function getContentWidth() {
+function getViewportDimensions() {
   // window.innerWidth to get the scrollable size of the window (irrespective of zoom)
   // window.outerWidth to get the size of the visible area
   // window.devicePixelRatio to get ratio of logical pixels to physical pixels
   return Promise.resolve({
-    scrollWidth: window.innerWidth,
-    viewportWidth: window.outerWidth,
+    scrollPosition: {x: window.scrollX, y: window.scrollY},
+    innerWidth: window.innerWidth,
+    innerHeight: window.innerHeight,
+    outerWidth: window.outerWidth,
+    outerHeight: window.outerHeight,
     devicePixelRatio: window.devicePixelRatio,
   });
 }
 
-class ContentWidth extends Gatherer {
+class ViewportDimensions extends Gatherer {
 
   /**
    * @param {!Object} options
-   * @return {!Promise<{scrollWidth: number, viewportWidth: number, devicePixelRatio: number}>}
+   * @return {!Promise<{innerWidth: number, outerWidth: number, devicePixelRatio: number}>}
    */
   afterPass(options) {
     const driver = options.driver;
 
-    return driver.evaluateAsync(`(${getContentWidth.toString()}())`)
+    return driver.evaluateAsync(`(${getViewportDimensions.toString()}())`)
 
     .then(returnedValue => {
-      if (!Number.isFinite(returnedValue.scrollWidth) ||
-          !Number.isFinite(returnedValue.viewportWidth) ||
+      if (!Number.isFinite(returnedValue.innerWidth) ||
```

Comment:
time for something more succinct here? No `Object.values` yet, but could do like
```js
const allNumeric = Object.keys(returnedValue).every(key => Number.isFinite(returnedValue[key]));
if (!allNumeric) {
  // ...
}
```
?